Userland/Applications/Calendar/Calendar.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Calendar.h"
+#include <AK/JsonParser.h>
+#include <LibConfig/Client.h>
+#include <LibFileSystemAccessClient/Client.h>
+
+namespace Calendar {
+
+Calendar::Calendar()
+{
+}
+
+OwnPtr<Calendar> Calendar::create()
+{
+    return adopt_own(*new Calendar());
+}
+
+ErrorOr<void> Calendar::add_event(JsonObject event)
+{
+    if (!m_events.has_value()) {
+        m_events = TRY(AK::JsonParser("[]"sv).parse()).as_array();
+    }
+
+    TRY(m_events->append(move(event)));
+
+    set_dirty(true);
+
+    on_events_change();
+
+    return {};
+}
+
+void Calendar::set_events(JsonArray events)
+{
+    m_events = events;
+    on_events_change();
+}
+
+void Calendar::save(FileSystemAccessClient::File& file)
+{
+    auto filename = file.filename().to_deprecated_string();
+    set_filename(filename);
+    set_dirty(false);
+
+    auto stream = file.release_stream();
+    if (m_events.has_value()) {
```

Comment:
the code would have less conditionals on events having value or not… if it just have it—consider initialising `m_events` as `JsonArray` (ditch the `Optional`) to avoid checks in `save`, `add_events` etc.